src/Compilers/CSharp/Test/Syntax/Parsing/RawInterpolatedStringLiteralCompilingTests.cs

```diff
@@ -0,0 +1,1757 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Parsing;
+
+public class RawInterpolatedStringLiteralCompilingTests : CompilingTestBase
+{
+    private static string Render(string markup, string? normalize)
+    {
+        markup = markup.Replace('␠', ' ').Replace('␉', '\t');
+
+        // If we're normalizing newlines, convert everything to \n, then convert that to the newline form asked for.
+        if (normalize != null)
+        {
+            markup = markup.Replace("\r\n", "\n");
+            markup = markup.Replace("\r", "\n");
+            markup = markup.Replace("\n", normalize);
+        }
+
+        return markup;
+    }
+
+    private void RenderAndVerify(string markup, string expectedOutput)
+    {
+        RenderAndVerify(markup, expectedOutput, normalize: null);
+        RenderAndVerify(markup, expectedOutput, normalize: "\r\n");
+        RenderAndVerify(markup, expectedOutput, normalize: "\n");
+        RenderAndVerify(markup, expectedOutput, normalize: "\r");
+    }
+
+    private void RenderAndVerify(string markup, string expectedOutput, string? normalize)
+    {
+        var text = Render(markup, normalize);
+        ParseAllPrefixes(text);
+        CompileAndVerify(text, expectedOutput: Render(expectedOutput, normalize), trimOutput: false);
+    }
+
+    private static void RenderAndVerify(string markup, params DiagnosticDescription[] expected)
+    {
+        RenderAndVerify(markup, expected, normalize: null);
+        RenderAndVerify(markup, expected, normalize: "\r\n");
+        RenderAndVerify(markup, expected, normalize: "\n");
+        RenderAndVerify(markup, expected, normalize: "\r");
+    }
+
+    private static void RenderAndVerify(string markup, DiagnosticDescription[] expected, string? normalize)
+    {
+        var text = Render(markup, normalize);
+        ParseAllPrefixes(text);
+        CreateCompilation(text).VerifyDiagnostics(expected);
+    }
+
+    private static void ParseAllPrefixes(string text)
+    {
+        // ensure the parser doesn't crash on any test cases.
+        for (var i = 0; i < text.Length; i++)
+            SyntaxFactory.ParseCompilationUnit(text[0..^i]);
+    }
+
+    [Fact]
+    public void TestDownlevel()
+    {
+        CreateCompilation(
+@"class C
+{
+    const string s = $"""""" """""";
+}", parseOptions: TestOptions.Regular10).VerifyDiagnostics(
```

Comment:
I couldn't find any test that references CSharpNext or RegularNext. Tests that are intended to cover LangVer check should use those instead of default (which is "preview").

Reply:
i'm not sure what you mean here.  this test is specifically trying to validate that we get an unsupported lang error if you try to use these in C# 10 or below.  If i use RegularNext then i won't get an error here as this feature is supported in c#-preview.

Reply:
We should have LangVer tests that uses C# 11 specifically and is locked on that version (as opposed to "preview" which always slides forward with releases).
The way to do that is to use CSharpNext/RegularNext, which we will replace with CSharp11/Regular11 once those become available.

Reply:
but i'm not referencing 'preview' here... 

> The way to do that is to use CSharpNext/RegularNext, which we will replace with CSharp11/Regular11 once those become available.

But that would be bad here.  Then the error woudl go away.  But this tests for the presence of the error.

Reply:
Note: this is the only comment I'd consider blocking for this PR. A test with CSharpNext for raw string literal and raw string interpolation. 
For other suggestions, it's at your discretion.

Reply:
talked offline.  confusion averted :)